fuse_core/test/test_throttled_callback.cpp

```diff
@@ -78,41 +78,60 @@ class PointPublisher
     ros::Rate rate(frequency_);
     for (size_t i = 0; i < num_messages; ++i)
     {
-      geometry_msgs::Point point_message;
-      point_message.x = i;
+      geometry_msgs::PointStamped point_message;
+      point_message.point.x = i;
 
       publisher_.publish(point_message);
 
       rate.sleep();
     }
   }
 
+  void publishWithHeaderTimestamp(const ros::Time timestamp)
+  {
+    ros::WallTime subscriber_timeout = ros::WallTime::now() + ros::WallDuration(1.0);
+    while (publisher_.getNumSubscribers() < 1u && ros::WallTime::now() < subscriber_timeout)
+    {
+      ros::WallDuration(0.01).sleep();
+    }
+
+    ASSERT_GE(publisher_.getNumSubscribers(), 1u);
+
+    geometry_msgs::PointStamped point_message;
+    point_message.header.stamp = timestamp;
+
+    publisher_.publish(point_message);
+  }
+
 private:
   ros::NodeHandle node_handle_;  //!< The node handle
   ros::Publisher publisher_;     //!< The publisher
   double frequency_{ 10.0 };     //!< The publish rate frequency
 };
 
 /**
- * @brief A dummy point sensor model that uses a fuse_core::ThrottledMessageCallback<geometry_msgs::Point> with a keep
+ * @brief A dummy point sensor model that uses a fuse_core::ThrottledMessageCallback<geometry_msgs::PointStamped> with a keep
  * and drop callback.
  *
  * The callbacks simply count the number of times they are called, for testing purposes. The keep callback also caches
  * the last message received, also for testing purposes.
  */
+
+ static_assert(ros::message_traits::HasHeader<geometry_msgs::PointStamped>::value == true);
```

Comment:
```suggestion
static_assert(ros::message_traits::HasHeader<geometry_msgs::PointStamped>::value == true);
```

I didn't know you could have this here. I thought it had to be in a function or method scope.

Reply:
This is how I usually use them. I kind of prefer this to `std::enable_if` which is cool but smells a little bit too much like magic to me.